### PR TITLE
Ensure we have the latest fnproject/dind

### PR DESCRIPTION
**Pull request by jan-g**
*To release this:*

Ensure that fnproject/dind:17.12 is rebuilt *and published* first.

At that point the release process should grab it and build a new fnproject/fnserver based off it.

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -26,6 +26,7 @@ perl -i -pe 's/\d+\.\d+\.\K(\d+)/$1+1/e' $version_file
 version=$(grep -m1 -Eo "[0-9]+\.[0-9]+\.[0-9]+" $version_file)
 echo "Version: $version"
 
+docker pull fnproject/dind:17.12
```
Comment (treeder) [inline release.sh]: It will always get the latest dind during release already, this is probably unecessary.

Comment (rdallman) [reply release.sh]: +1 to remove, we have https://github.com/fnproject/fn/blob/master/Dockerfile#L9 -- this line risks going stale vs the Dockerfile and we'd eat the pull

Comment (jan-g) [reply release.sh]: While staleness is a problem, are we guaranteed that the docker caches on the ci machines are cleared between runs?

If so, I've no problem with dropping this. On t'other hand, if it doesn't *and* if we shift to "rebuild these tools when changed" then we risk a change on a branch leaving a broken dind behind for main to consume.

Comment (jan-g) [reply release.sh]: ... rather than ask I'll check it :-)

Comment (rdallman) [reply release.sh]: with circleci they make you pay to have caching, and we're not ponying up, so I'm pretty sure the caches are cleared out (it has been a bit of a thorn, to date)

Comment (skinowski) [reply release.sh]: We can explicitly clear them in Makefile (eg. clear-images target) and hook that up to circleci/config.yaml.

**Pull request merged**